src/dune_engine/scheduler.ml

```diff
@@ -172,6 +177,7 @@ end = struct
     | Job_completed of job * Unix.process_status
     | Signal of Signal.t
     | Rpc of Fiber.fill
+    | Jobserver of Fiber.fill
```

Comment:
`Rpc` and `Jobserver` seems to be exactly the same. What about replacing them both by a single `Thread_safe_ivar` event?